src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Collections/ImmutableList.cs

```diff
@@ -296,8 +294,10 @@ public IImmutableList<T> Replace(T oldValue, T newValue, IEqualityComparer<T> eq
 		public ImmutableList<T> Add(T value)
 		{
 			var newData = new T[_data.Length + 1];
-			Array.Copy(_data, newData, _data.Length);
-			newData[_data.Length] = value;
+			Span<T> newDataSpan = MemoryMarshal.CreateSpan(ref MemoryMarshal.GetArrayDataReference(newData), newData.Length);
+			Span<T> dataSpan = MemoryMarshal.CreateSpan(ref MemoryMarshal.GetArrayDataReference(_data), _data.Length); ;
```

Comment:
nit: the CI might fail because of this
```suggestion
			Span<T> dataSpan = MemoryMarshal.CreateSpan(ref MemoryMarshal.GetArrayDataReference(_data), _data.Length);
```